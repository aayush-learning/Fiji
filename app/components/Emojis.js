export const people = [ "😀", "😬", "😁", "😂", "😃", "😄", "😅", "😆", "😇", "😉", "😊",
                        "☺", "😋", "😌", "😍", "😘", "😗", "😙", "😚",
                        "😜", "😝", "😛", "😎", "😏", "😶", "😐",
                        "😑", "😒", "😳", "😞", "😟", "😠", "😡", "😔",
                        "😕", "☹", "😣", "😖", "😫", "😩", "😤", "😮", "😱",
                        "😨", "😰", "😯", "😦", "😧", "😢", "😥", "😪", "😓", "😭",
                        "😵", "😲", "😷", "😴", "💤", "💩", "😈",
                        "👿", "👹", "👺", "💀", "👻", "👽", "😺", "😸", "😹",
                        "😻", "😼", "😽", "🙀", "😿", "😾", "🙌", //"🙌🏻", "🙌🏼", "🙌🏽",
                        "👯", "👫", "👬", "👭", "🙇", //"🙇🏻", "🙇🏼", "🙇🏽", "🙇🏾", "🙇🏿",
                        "💁", //"💁🏻", "💁🏼", "💁🏽", "💁🏾", "💁🏿",
                        "🙅", //"🙅🏻", "🙅🏼", "🙅🏽", "🙅🏾", "🙅🏿",
                        "🙆", //"🙆🏻", "🙆🏼", "🙆🏽", "🙆🏾", "🙆🏿",
                        "🙋", //"🙋🏻",
                        "💘", "💝"];

export const animals = ["🐶", "🐱", "🐭", "🐹", "🐰", "🐻", "🐼", "🐨", "🐯", "🐮",
                    "🐷", "🐽", "🐸", "🐙", "🐵", "🙈", "🙉", "🙊", "🐒", "🐔",
                    "🐧", "🐦", "🐤", "🐣", "🐥", "🐺", "🐗", "🐴", "🐝",
                    "🐛", "🐌", "🐞", "🐜", "🐍", "🐢", "🐠",
                    "🐟", "🐡", "🐬", "🐳", "🐋", "🐊", "🐆", "🐅", "🐃", "🐂",
                    "🐄", "🐪", "🐫", "🐘", "🐐", "🐏", "🐑", "🐎", "🐖", "🐀",
                    "🐁", "🐓", "🐕", "🐩", "🐈", "🐇", "🐾",
                    "🐉", "🐲", "🌵", "🎄", "🌲", "🌳", "🌴", "🌱", "🌿", "☘",
                    "🍀", "🎍", "🎋", "🍃", "🍂", "🍁", "🌾", "🌺", "🌻", "🌹",
                    "🌷", "🌼", "🌸", "💐", "🍄", "🌰", "🎃", "🐚", "🌎",
                    "🌍", "🌏", "🌕", "🌖", "🌗", "🌘", "🌑", "🌒", "🌓", "🌔",
                    "🌚", "🌝", "🌛", "🌜", "🌞", "🌙", "⭐", "🌟", "💫", "✨",
                    "☄", "☀", "⛅", "☁",
                    "⚡", "🔥", "💥", "❄", "☃", "⛄", "💨",
                    "☂", "☔", "💧", "💦", "🌊"];

export const food = ["🍏", "🍎", "🍐", "🍊", "🍋", "🍌", "🍉", "🍇", "🍓", "🍈", "🍒",
                    "🍑", "🍍", "🍅", "🍆",  "🌽", "🍠", "🍯", "🍞",
                    "🍗", "🍖", "🍤", "🍳", "🍔", "🍟", "🍕", "🍝",
                    "🍜", "🍲", "🍥", "🍣", "🍱", "🍛", "🍙", "🍚", "🍘",
                    "🍢", "🍡", "🍧", "🍨", "🍦", "🍰", "🎂", "🍮", "🍬", "🍭",
                    "🍫", "🍩", "🍪", "🍺", "🍻", "🍷", "🍸", "🍹",
                    "🍶", "🍵", "☕", "🍼", "🍴"];

export const activity = emojiStringToArray('⚽️ 🏀 🏈 🎾 🏉 🎱 🎿 🏂 🎣 🚣 🏊 🏄 🛀 🚴 🚵 🏇 🏆 🎽 🎫 🎭 🎨 🎪 🎤 🎧 🎼 🎹 🎷 🎺 🎸 🎻 🎬 🎮 👾 🎯 🎲 🎰 🎳');
export const travel = emojiStringToArray('🚗🚕 🚙 🚌 🚎 🚓 🚑 🚒 🚐 🚚 🚛 🚜 🚲 🚨 🚔 🚍 🚘 🚖 🚡 🚠 🚟 🚃 🚋 🚝 🚄 🚅 🚈 🚞 🚂 🚆 🚇 🚊 🚉 🚁 🚤 🚀 💺 🚧 🚏 🚦 🚥 🏁 🚢 🎡 🎢 🎠 🌁 🗼 🏭 🎑 🗻 🌋 🗾 🌅 🌄 🌇 🌆 🌃 🌉 🌌 🌠 🎇 🎆 🌈 🏰 🏯 🗽 🏠 🏡 🏢 🏬 🏣 🏤 🏥 🏦 🏨 🏪 🏫 🏩 💒');
export const objects = emojiStringToArray('⌚️ 📱 📲 💻 ⌨ 💽 💾 💿 📀 📼 📷 📹 🎥 📞 📟 📠 📺 📻 ⏰ ⏳ ⌛️ 📡 🔋 🔌 💡 🔦 💸 💵 💴 💶 💷 💰 💳 💎 ⚖ 🔧 🔨 ⚒ 🔩 ⚙ 🔫 💣 🔪 ⚔ 🚬 ☠ ⚰ ⚱ 🔮 💈 ⚗ 🔭 🔬 💊 💉 🔖 🚽 🚿 🛁 🔑 🚪 🗿 🎈 🎏 🎀 🎁 🎊 🎉 🎎 🎐 🎌 🏮 📩 📨 📧 💌 📮 📪 📫 📬 📭 📦 📯 📥 📤 📜 📃 📑 📊 📈 📉 📄 📅 📆 📇 📋 📁 📂 📰 📓 📕 📗 📘 📙 📔 📒 📚 📖 🔗 📎 📐 📏 📌 📍 🚩 🔐 🔒 🔓 🔏 📝 🔍 🔎');
// export const symbols = emojiStringToArray('💛 💙 💜 💔 💕 💞 💓 💗 💖 💘 💝 💟 ☮ ☪ ☸ 🔯 ☦ ⛎ 🆔 ⚛ 🈳 🈹 ☢ ☣ 📴 📳 🈶 🈚️ 🈸 🈺 🈷️ 🆚 🉑 💮 🉐 ㊙️ ㊗️ 🈴 🈵 🈲 🅰️ 🅱️ 🆎 🆑 🅾️ 🆘 ⛔️ 📛 🚫 ❌ ⭕️ 💢 🚷 🚯 🚳 🚱 🔞 📵 ❗️ ❕ ❓ ❔ 💯 🔅 🔆 🔱 ⚜ 〽️ 🚸 🔰 🈯️ 💹 ❎ ✅ 💠 🌀 ➿ 🌐 Ⓜ️ 🏧 🈂️ 🛂 🛃 🛄 🛅 🚭 🚾 🅿️ 🚰 🚹 🚺 🚼 🚻 🚮 🎦 📶 🈁 🆖 🆗 🆙 🆒 🆕 🆓 🔟 🔢 ⏩ ⏪ 🔀 🔁 🔂 🔼 🔽 ⏫ ⏬ ⬅️ ⬆️ ⬇️ 🔄 ⤴️ ⤵️ 🔤 🔡 🔠 🔣 🎵 🎶 〰️ ➰ 🔃 ➕ ➖ ➗ 💲 💱 🔚 🔙 🔛 🔝 🔜 🔘 ⚪️ ⚫️ 🔴 🔵 🔸 🔹 🔶 🔷 🔺 ⬛️ ⬜️ 🔻 🔲 🔳 🔈 🔉 🔊 🔇 📣 📢 🔔 🔕 🃏 🀄️ 💭 💬 🕐 🕑 🕒 🕓 🕔 🕕 🕖 🕗 🕘 🕙 🕚 🕛 🕜 🕝 🕞 🕟 🕠 🕡 🕢 🕣 🕤 🕥 🕦 🕧');
// export const flags = emojiStringToArray('');

function emojiStringToArray(str) {
  // split = str.split(/([\uD800-\uDBFF][\uDC00-\uDFFF])/);
  split = str.split(' ')
  arr = [];
  for (var i=0; i<split.length; i++) {
    char = split[i]
    if (char !== "" && char !== ' ') {
      arr.push(char);
    }
  }
  return arr;
};
