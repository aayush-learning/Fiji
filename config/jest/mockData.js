import { fetchUsersFailure } from '../../app/redux/users';

const contentJson = require('../seed/content.json')
const usersJson = require('../seed/users.json')
const userAliceJson = require('../seed/user_alice.json')
const storyJson = require('../seed/story.json')

export const titles = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('storytitle:')
})

export const stories = storyJson['docs'].filter(function (doc) {
  return doc._id.startsWith('story:')
})

export const games = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('game:')
})

export const gameThemeData = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('gametheme:')
})

export const stickersData = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('sticker:caterpillar:')
})

export const stickerPacksData = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('sticker-pack:')
})

export const highScoresData = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('high-score:')
})

export const story_alice_in_wonderland = storyJson['docs'].filter(function (doc) {
  return doc._id == 'story:alice-in-wonderland'
})[0]

export const users = usersJson['docs']

export const messages = userAliceJson['docs'].filter(function(doc) {
  return doc._id.startsWith('chat:mad-hatter:')
})