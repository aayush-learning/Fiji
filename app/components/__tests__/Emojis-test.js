import 'react-native';
import React from 'react';
import * as Emojis from '../Emojis';
import { users } from '../../../config/jest/mockData'

import renderer from 'react-test-renderer';

//npm test -- --updateSnapshot
//this above command is used to update the snapshot in windows

//npm test -- --coverage
//this above command is used to see the total coverage of the tested file in windows


//jest --updateSnapshot
//this above command is used to update the snapshot in IOS

//jest --coverage
//this above command is used to see the total coverage of the tested file in IOS

//npm test -- --coverage User-test
//this above command is used to test indivisual component

it('this is Emojis-test', () => {
  expect(Emojis).toMatchSnapshot();
});

