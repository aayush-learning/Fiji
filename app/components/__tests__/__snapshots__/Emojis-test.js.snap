// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`this is Emojis-test 1`] = `
Object {
  "activity": Array [
    "⚽️",
    "🏀",
    "🏈",
    "🎾",
    "🏉",
    "🎱",
    "🎿",
    "🏂",
    "🎣",
    "🚣",
    "🏊",
    "🏄",
    "🛀",
    "🚴",
    "🚵",
    "🏇",
    "🏆",
    "🎽",
    "🎫",
    "🎭",
    "🎨",
    "🎪",
    "🎤",
    "🎧",
    "🎼",
    "🎹",
    "🎷",
    "🎺",
    "🎸",
    "🎻",
    "🎬",
    "🎮",
    "👾",
    "🎯",
    "🎲",
    "🎰",
    "🎳",
  ],
  "animals": Array [
    "🐶",
    "🐱",
    "🐭",
    "🐹",
    "🐰",
    "🐻",
    "🐼",
    "🐨",
    "🐯",
    "🐮",
    "🐷",
    "🐽",
    "🐸",
    "🐙",
    "🐵",
    "🙈",
    "🙉",
    "🙊",
    "🐒",
    "🐔",
    "🐧",
    "🐦",
    "🐤",
    "🐣",
    "🐥",
    "🐺",
    "🐗",
    "🐴",
    "🐝",
    "🐛",
    "🐌",
    "🐞",
    "🐜",
    "🐍",
    "🐢",
    "🐠",
    "🐟",
    "🐡",
    "🐬",
    "🐳",
    "🐋",
    "🐊",
    "🐆",
    "🐅",
    "🐃",
    "🐂",
    "🐄",
    "🐪",
    "🐫",
    "🐘",
    "🐐",
    "🐏",
    "🐑",
    "🐎",
    "🐖",
    "🐀",
    "🐁",
    "🐓",
    "🐕",
    "🐩",
    "🐈",
    "🐇",
    "🐾",
    "🐉",
    "🐲",
    "🌵",
    "🎄",
    "🌲",
    "🌳",
    "🌴",
    "🌱",
    "🌿",
    "☘",
    "🍀",
    "🎍",
    "🎋",
    "🍃",
    "🍂",
    "🍁",
    "🌾",
    "🌺",
    "🌻",
    "🌹",
    "🌷",
    "🌼",
    "🌸",
    "💐",
    "🍄",
    "🌰",
    "🎃",
    "🐚",
    "🌎",
    "🌍",
    "🌏",
    "🌕",
    "🌖",
    "🌗",
    "🌘",
    "🌑",
    "🌒",
    "🌓",
    "🌔",
    "🌚",
    "🌝",
    "🌛",
    "🌜",
    "🌞",
    "🌙",
    "⭐",
    "🌟",
    "💫",
    "✨",
    "☄",
    "☀",
    "⛅",
    "☁",
    "⚡",
    "🔥",
    "💥",
    "❄",
    "☃",
    "⛄",
    "💨",
    "☂",
    "☔",
    "💧",
    "💦",
    "🌊",
  ],
  "food": Array [
    "🍏",
    "🍎",
    "🍐",
    "🍊",
    "🍋",
    "🍌",
    "🍉",
    "🍇",
    "🍓",
    "🍈",
    "🍒",
    "🍑",
    "🍍",
    "🍅",
    "🍆",
    "🌽",
    "🍠",
    "🍯",
    "🍞",
    "🍗",
    "🍖",
    "🍤",
    "🍳",
    "🍔",
    "🍟",
    "🍕",
    "🍝",
    "🍜",
    "🍲",
    "🍥",
    "🍣",
    "🍱",
    "🍛",
    "🍙",
    "🍚",
    "🍘",
    "🍢",
    "🍡",
    "🍧",
    "🍨",
    "🍦",
    "🍰",
    "🎂",
    "🍮",
    "🍬",
    "🍭",
    "🍫",
    "🍩",
    "🍪",
    "🍺",
    "🍻",
    "🍷",
    "🍸",
    "🍹",
    "🍶",
    "🍵",
    "☕",
    "🍼",
    "🍴",
  ],
  "objects": Array [
    "⌚️",
    "📱",
    "📲",
    "💻",
    "⌨",
    "💽",
    "💾",
    "💿",
    "📀",
    "📼",
    "📷",
    "📹",
    "🎥",
    "📞",
    "📟",
    "📠",
    "📺",
    "📻",
    "⏰",
    "⏳",
    "⌛️",
    "📡",
    "🔋",
    "🔌",
    "💡",
    "🔦",
    "💸",
    "💵",
    "💴",
    "💶",
    "💷",
    "💰",
    "💳",
    "💎",
    "⚖",
    "🔧",
    "🔨",
    "⚒",
    "🔩",
    "⚙",
    "🔫",
    "💣",
    "🔪",
    "⚔",
    "🚬",
    "☠",
    "⚰",
    "⚱",
    "🔮",
    "💈",
    "⚗",
    "🔭",
    "🔬",
    "💊",
    "💉",
    "🔖",
    "🚽",
    "🚿",
    "🛁",
    "🔑",
    "🚪",
    "🗿",
    "🎈",
    "🎏",
    "🎀",
    "🎁",
    "🎊",
    "🎉",
    "🎎",
    "🎐",
    "🎌",
    "🏮",
    "📩",
    "📨",
    "📧",
    "💌",
    "📮",
    "📪",
    "📫",
    "📬",
    "📭",
    "📦",
    "📯",
    "📥",
    "📤",
    "📜",
    "📃",
    "📑",
    "📊",
    "📈",
    "📉",
    "📄",
    "📅",
    "📆",
    "📇",
    "📋",
    "📁",
    "📂",
    "📰",
    "📓",
    "📕",
    "📗",
    "📘",
    "📙",
    "📔",
    "📒",
    "📚",
    "📖",
    "🔗",
    "📎",
    "📐",
    "📏",
    "📌",
    "📍",
    "🚩",
    "🔐",
    "🔒",
    "🔓",
    "🔏",
    "📝",
    "🔍",
    "🔎",
  ],
  "people": Array [
    "😀",
    "😬",
    "😁",
    "😂",
    "😃",
    "😄",
    "😅",
    "😆",
    "😇",
    "😉",
    "😊",
    "☺",
    "😋",
    "😌",
    "😍",
    "😘",
    "😗",
    "😙",
    "😚",
    "😜",
    "😝",
    "😛",
    "😎",
    "😏",
    "😶",
    "😐",
    "😑",
    "😒",
    "😳",
    "😞",
    "😟",
    "😠",
    "😡",
    "😔",
    "😕",
    "☹",
    "😣",
    "😖",
    "😫",
    "😩",
    "😤",
    "😮",
    "😱",
    "😨",
    "😰",
    "😯",
    "😦",
    "😧",
    "😢",
    "😥",
    "😪",
    "😓",
    "😭",
    "😵",
    "😲",
    "😷",
    "😴",
    "💤",
    "💩",
    "😈",
    "👿",
    "👹",
    "👺",
    "💀",
    "👻",
    "👽",
    "😺",
    "😸",
    "😹",
    "😻",
    "😼",
    "😽",
    "🙀",
    "😿",
    "😾",
    "🙌",
    "👯",
    "👫",
    "👬",
    "👭",
    "🙇",
    "💁",
    "🙅",
    "🙆",
    "🙋",
    "💘",
    "💝",
  ],
  "travel": Array [
    "🚗🚕",
    "🚙",
    "🚌",
    "🚎",
    "🚓",
    "🚑",
    "🚒",
    "🚐",
    "🚚",
    "🚛",
    "🚜",
    "🚲",
    "🚨",
    "🚔",
    "🚍",
    "🚘",
    "🚖",
    "🚡",
    "🚠",
    "🚟",
    "🚃",
    "🚋",
    "🚝",
    "🚄",
    "🚅",
    "🚈",
    "🚞",
    "🚂",
    "🚆",
    "🚇",
    "🚊",
    "🚉",
    "🚁",
    "🚤",
    "🚀",
    "💺",
    "🚧",
    "🚏",
    "🚦",
    "🚥",
    "🏁",
    "🚢",
    "🎡",
    "🎢",
    "🎠",
    "🌁",
    "🗼",
    "🏭",
    "🎑",
    "🗻",
    "🌋",
    "🗾",
    "🌅",
    "🌄",
    "🌇",
    "🌆",
    "🌃",
    "🌉",
    "🌌",
    "🌠",
    "🎇",
    "🎆",
    "🌈",
    "🏰",
    "🏯",
    "🗽",
    "🏠",
    "🏡",
    "🏢",
    "🏬",
    "🏣",
    "🏤",
    "🏥",
    "🏦",
    "🏨",
    "🏪",
    "🏫",
    "🏩",
    "💒",
  ],
}
`;
